met-web/tests/unit/components/engagementForm.test.tsx

```diff
@@ -1,15 +1,142 @@
+        });
+        expect(screen.getByText('Create Engagement Draft')).toBeInTheDocument();
+    });
+
+    test('Test cannot create engagement with empty fields', async () => {
+        useParamsMock.mockReturnValue({ engagementId: 'create' });
+        const { container, getByText } = render(<EngagementForm />);
+
+        const createButton = getByText('Create Engagement Draft');
+        fireEvent.click(createButton);
+
+        expect(container.querySelectorAll('.Mui-error').length).toBeGreaterThan(0);
+    });
+
+    test('Engagement form with saved engagement should display saved info', async () => {
+        useParamsMock.mockReturnValue({ engagementId: '1' });
+        render(<EngagementForm />);
+
+        await waitFor(() => {
+            expect(screen.getByDisplayValue('Test Engagement')).toBeInTheDocument();
+        });
+
+        expect(getEngagementMock).toHaveBeenCalledOnce();
+        expect(screen.getByText('Update Engagement')).toBeInTheDocument();
+        expect(screen.getByDisplayValue('2022-09-01')).toBeInTheDocument();
+        expect(screen.getByDisplayValue('2022-09-30')).toBeInTheDocument();
+        expect(screen.getByText('Survey 1')).toBeInTheDocument();
+    });
+
+    test('Update engagement button should trigger Put call', async () => {
+        useParamsMock.mockReturnValue({ engagementId: '1' });
+        render(<EngagementForm />);
+
+        await waitFor(() => {
+            expect(screen.getByDisplayValue('Test Engagement')).toBeInTheDocument();
+        });
+        const updateButton = screen.getByText('Update Engagement');
+
+        fireEvent.click(updateButton);
+
+        await waitFor(() => {
+            expect(putEngagementMock).toHaveBeenCalledOnce();
+        });
+    });
+
+    test('Cannot add survey to unsaved engagement', async () => {
+        const openNotificationMock = jest.spyOn(notificationSlice, 'openNotification').mockImplementation(jest.fn());
+        useParamsMock.mockReturnValue({ engagementId: 'create' });
+        render(<EngagementForm />);
+
+        const addSurveyButton = screen.getByText('Add Survey');
+
+        fireEvent.click(addSurveyButton);
+
+        expect(openNotificationMock).toHaveBeenNthCalledWith(1, {
+            severity: 'error',
+            text: 'Please save the engagement before adding a survey',
+        });
+        openNotificationMock.mockClear();
+    });
+
+    test('Modal with warning appears when removing survey', async () => {
+        useParamsMock.mockReturnValue({ engagementId: '1' });
+        render(<EngagementForm />);
+
+        await waitFor(() => {
+            expect(screen.getByDisplayValue('Test Engagement')).toBeInTheDocument();
+        });
+
+        const removeSurveyButton = screen.getByTestId('survey-widget/remove');
+
+        fireEvent.click(removeSurveyButton);
+
+        expect(screen.getByText('Do you want to remove this survey?')).toBeInTheDocument();
```

Comment:
It might be worth having an id for things like this too, just bc i see texts are usually fragile, the change often and any change in the wording will cause the test to fail, the same goes for the notification above, I think in general the tests should be agnostic of the contents and more focused on the elements itself

Reply:
I see your point here, we don't want to get by something that might potentially change. My thoughts here was I wanted to minimize changing the code as much as possible. If I don't have to go to the code and add a test-id then I won't. In this particular case even If I get by id I still need to check the message and make sure it's a warning message. If the modal is no longer a 'warning' modal this test should fail and be updated right?

Reply:
That’s right!

Reply:
getByText is sometimes the closest thing to 'get by semantic' I could use

Reply:
Overal it’s looking really good! Good job!